pkg/3scale/amp/operator/zync_reconciler.go

```diff
@@ -206,6 +215,153 @@ func (r *ZyncReconciler) Reconcile() (reconcile.Result, error) {
 	return reconcile.Result{}, nil
 }
 
+// deleteZyncComponents handles the removal of all zync components
+// The operator only enters this function when the APIManager .spec.zync.enabled was initially set to true but then changed to false
+// When this happens, the operator will call this function on each reconcile loop to ensure the resources are cleaned up
+func (r *ZyncReconciler) deleteZyncComponents(zync *component.Zync, ampImages *component.AmpImages) error {
+	// ZyncQue PrometheusRules
+	zyncQuePrometheusRules := zync.ZyncPrometheusRules()
+	common.TagObjectToDelete(zyncQuePrometheusRules)
+	err := r.ReconcilePrometheusRules(zyncQuePrometheusRules, reconcilers.CreateOnlyMutator)
```

Comment:
A bit confused about this...
So I understand what we want to do here, but does it really work if it's a CreateOnlyMutator ?!
I mean..so if we have prometheusRules, and then we tag it but run it through CreateOnlyMutator will it really delete the rule?

Reply:
The deletion does work as intended even when using the CreateOnlyMutator but I agree it's counter intuitive. I'll create a DeleteMutator to make it more clear.

Reply:
I think that is fine if it works, I would leave it.

Just thinking out loud here...would not be easier to tag all zync objects with zync deployment as the owner so we leverage garbage collector?

However, although it would make cleanup significantly easier, is this creating a risk of non-working zync if someone decides just to nuke the deployment and have operator re-creating it?

WDYT?

Reply:
just to be clear, not a big deal to leave it as it is as well..

Reply:
Fixed in latest commit